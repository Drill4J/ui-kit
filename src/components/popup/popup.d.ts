/// <reference types="react" />
interface Props {
    className?: string;
    children: React.ReactChild;
    header: React.ReactNode;
    onToggle: (isOpen: boolean) => void;
    isOpen: boolean;
    type?: 'info' | 'error';
    closeOnFadeClick?: boolean;
}
export declare const Popup: {
    ({ className, header, children, onToggle, isOpen, type, closeOnFadeClick, }: Props): JSX.Element;
    displayName: string;
    __docgenInfo: {
        description: string;
        displayName: string;
        props: {
            className: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            header: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            onToggle: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            isOpen: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            type: {
                defaultValue: {
                    value: string;
                };
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            closeOnFadeClick: {
                defaultValue: {
                    value: boolean;
                };
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
        };
    };
};
export {};
