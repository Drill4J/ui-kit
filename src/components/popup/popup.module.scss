@import '../../style-guide/index';

.popup {
  display: contents;

  &__header {
    width: auto;
    min-height: 64px;
    padding: 0 24px;
    border-bottom: 1px solid map-get($monochrome, medium-tint);
    font-size: 20px;
    line-height: 32px;
    color: map-get($monochrome, black);
  }

  &__close-button {
    cursor: pointer;
  }

  &__content {
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: 100;
    background-color: map-get($monochrome, white);
    transform: translate(-50%, -50%);

    &--type {
      &--info {
        box-shadow: 0 -4px 0 0 map-get($primary-blue, default), 0 0 24px 0 rgba(0, 0, 0, 0.15);
      }
      &--error {
        box-shadow: 0 -4px 0 0 map-get($red, default), 0 0 24px 0 rgba(0, 0, 0, 0.15);
      }
    }
  }

  &__fade {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    opacity: 0.75;
    background-color: map-get($monochrome, white);
    z-index: 90;
  }
}
