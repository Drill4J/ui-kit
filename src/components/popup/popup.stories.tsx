import React from 'react';
import { ComponentMeta } from '@storybook/react';
import 'twin.macro';

import { Popup } from './popup';

export default {
  title: 'Popup',
  component: Popup,
  argTypes: {
    type: {
      control: {
        type: 'radio',
      },
      options: ['info', 'error'],
    },
  },
} as ComponentMeta<typeof Popup>;

const Template = (args) => (
  <Popup value={100} {...args}>
    <div tw="p-6">
      <h2>Error popup content</h2>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corrupti ullam praesentium laudantium delectus...</p>
      <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Repellendus, iure magni quia excepturi aliquid quas nostrum?</p>
      <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quas nisi corrupti dolorem incidunt quisquam.</p>
    </div>
  </Popup>
);
export const Default = Template.bind({});
