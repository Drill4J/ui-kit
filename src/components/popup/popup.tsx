import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Panel } from '../../layouts';
import { Portal } from '../portal';
import { Icons } from '../icon';

import styles from './popup.module.scss';

interface Props {
  className?: string;
  children: React.ReactChild;
  header: React.ReactNode;
  onToggle: (isOpen: boolean) => void;
  isOpen: boolean;
  type?: 'info' | 'error';
  closeOnFadeClick?: boolean;
}

const popup = BEM(styles);

export const Popup = popup(
  ({
    className,
    header,
    children,
    onToggle,
    isOpen,
    type = 'info',
    closeOnFadeClick = false,
  }: Props) => (
    <div className={className}>
      <Portal>
        {isOpen && (
          <div className={className}>
            <Content type={type}>
              <Header align="space-between">
                {header}
                <CloseButton onClick={() => onToggle(!isOpen)} />
              </Header>
              {children}
            </Content>
            <Fade onClick={() => closeOnFadeClick && onToggle(!isOpen)} />
          </div>
        )}
      </Portal>
    </div>
  ),
);

const Content = popup.content('div');
const Header = popup.header(Panel);
const CloseButton = popup.closeButton(Icons.Close);
const Fade = popup.fade('div');
