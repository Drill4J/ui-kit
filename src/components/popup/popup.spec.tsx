import renderer from 'react-test-renderer';

import { Popup } from './popup';

describe('Popup', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Popup
          isOpen={false}
          type="info"
          onToggle={() => {}}
          header={(<h1>Popup</h1>)}
          closeOnFadeClick
        >
          <div>
            <h2>Popup content</h2>
          </div>
        </Popup>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
