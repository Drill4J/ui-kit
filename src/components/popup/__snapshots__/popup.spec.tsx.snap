// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match snapshot 1`] = `
<div
  className="popup popup--type--info"
/>
`;
